### PR TITLE
Added copy parameters to enable parallel build

### PR DESCRIPTION
**Pull request by dashkodo**
- [x] You've read the [Contributor Guide](https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/dotnet/aspnetcore/blob/main/CODE-OF-CONDUCT.md).
- [ ] You've included unit or integration tests for your change, where applicable.
- [ ] You've included inline docs for your change, where applicable.
- [x] There's an open issue for the PR that you are making. If you'd like to propose a new feature or change, please open an issue to discuss the change or find an existing issue.

**PR Title**
Added parameters to Copy task that is executed for setting up correct dependency environment in the tests.

**PR Description**
When enabling multiple msbuild nodes it could fail due to locking "xxx.deps.json" files
We have faced with this issue using latest version of microsoft.aspnetcore.mvc.testing nuget package
C:\ZZZZ\nuget\microsoft.aspnetcore.mvc.testing\5.0.7\build\net5.0\Microsoft.AspNetCore.Mvc.Testing.targets(56, 5): error MSB3027: Unable to copy file "D:\XXXX\bin\Release\YYYY.deps.json" to "bin\Release\YYYY.deps.json". The process cannot access the file 'bin\Release\YYYY.deps.json' because it is being used by another process.

Addresses #33757

### CODE REVIEW

File: src/Mvc/Mvc.Testing/src/Microsoft.AspNetCore.Mvc.Testing.targets [hunk 1]
```diff
@@ -64,8 +64,17 @@
         Condition="'%(_ContentRootProjectReferences.Identity)' != ''"
         Include="$([System.IO.Path]::ChangeExtension('%(_ContentRootProjectReferences.ResolvedFrom)', '.deps.json'))" />
     </ItemGroup>
+    
+    <PropertyGroup>
+        <CreateHardLinksForMvcCopyDependencyFilesIfPossible Condition="'$(CreateHardLinksForMvcCopyDependencyFilesIfPossible)' == ''">false</CreateHardLinksForMvcCopyDependencyFilesIfPossible>
+        <CreateSymbolicLinksForMvcCopyDependencyFilesIfPossible Condition="'$(CreateSymbolicLinksMvcCopyDependencyFilesIfPossible)' == ''">false</CreateSymbolicLinksForMvcCopyDependencyFilesIfPossible>
+    </PropertyGroup>
```
Comment (javiercn) [inline src/Mvc/Mvc.Testing/src/Microsoft.AspNetCore.Mvc.Testing.targets:71]:
```suggestion
    <PropertyGroup>
        <_CreateHardLinksForMvcCopyDependencyFilesIfPossible Condition="'$(CreateHardLinksForMvcCopyDependencyFilesIfPossible)' == ''">false</_CreateHardLinksForMvcCopyDependencyFilesIfPossible>
        <_CreateSymbolicLinksForMvcCopyDependencyFilesIfPossible Condition="'$(CreateSymbolicLinksMvcCopyDependencyFilesIfPossible)' == ''">false</_CreateSymbolicLinksForMvcCopyDependencyFilesIfPossible>
    </PropertyGroup>
```

File: src/Mvc/Mvc.Testing/src/Microsoft.AspNetCore.Mvc.Testing.targets [hunk 2]
```diff
@@ -64,8 +64,17 @@
...
 
-    <Copy SourceFiles="%(DepsFilePaths.FullPath)" DestinationFolder="$(OutDir)" Condition="Exists('%(DepsFilePaths.FullPath)')" />
+    <Copy SourceFiles="%(DepsFilePaths.FullPath)"
+       DestinationFolder="$(OutDir)"
+       UseHardlinksIfPossible="$(CreateHardLinksForMvcCopyDependencyFilesIfPossible)"
+       UseSymboliclinksIfPossible="$(CreateSymbolicLinksForMvcCopyDependencyFilesIfPossible)"
+       Condition="Exists('%(DepsFilePaths.FullPath)')" />
```
Comment (javiercn) [inline src/Mvc/Mvc.Testing/src/Microsoft.AspNetCore.Mvc.Testing.targets:77]:
```suggestion
    <Copy SourceFiles="%(DepsFilePaths.FullPath)"
       DestinationFolder="$(OutDir)"
       UseHardlinksIfPossible="$(_CreateHardLinksForMvcCopyDependencyFilesIfPossible)"
       UseSymboliclinksIfPossible="$(_CreateSymbolicLinksForMvcCopyDependencyFilesIfPossible)"
       Condition="Exists('%(DepsFilePaths.FullPath)')" />
```

**Review by javiercn — CHANGES_REQUESTED** [review]
Thanks for the contribution.

I've made a suggestion since we normally use `_` for things we don't want people to depend on or configure individually (like this).

The default value should also come from the standard `CreateHardLinksIfPossible` (or however is named in copy files) since that's the single switch users turn on or off

**Comment by dashkodo** [comment]
I could not find any generict `CreateHardLinksIfPossible`
only specialized `CreateHardLinksFor_Something_IfPossible` are located in targets.

**Comment by pranavkm** [comment]
@dashkodo feel free to reopen this PR once you've had a chance to address this feedback:

> The default value should also come from the standard CreateHardLinksIfPossible (or however is named in copy files) since that's the single switch users turn on or off

**Comment by dashkodo** [comment]
Hi @pranavkm  I've added defaults. Hope it's matching your expectation

**Comment by dashkodo** [comment]
@javiercn Hi, Let me know if any further changes required.
BTW I have no permission to reopen pull request. Could you suggest what is the better here - create new one or open this?

**Pull request closed without merging**